test/jdk/sun/nio/cs/TestEncoderReplaceLatin1.java

```diff
@@ -0,0 +1,263 @@
+
+        // Create an encoder
+        CharsetEncoder encoder = createEncoder(charset);
+        if (encoder == null) {
+            return;
+        }
+
+        // Find an unmappable character to test the `REPLACE` action.
+        char[] unmappable = findUnmappable(encoder);
+        if (unmappable == null) {
+            return;
+        }
+
+        // Configure the `REPLACE` action
+        byte[] replacement = findCustomReplacement(encoder, new byte[]{(byte) unmappable[0]});
+        if (replacement == null) {
+            return;
+        }
+        encoder.onUnmappableCharacter(CodingErrorAction.REPLACE).replaceWith(replacement);
+
+        // Verify the replacement
+        System.err.println("Verifying replacement... " + Map.of(
+                "unmappable", TestEncoderReplaceLatin1.prettyPrintChars(unmappable),
+                "replacement", TestEncoderReplaceLatin1.prettyPrintBytes(replacement)));
+        testCharsetEncoderReplace(encoder, unmappable, replacement);
+        testArrayEncoderLatin1Replace(encoder, unmappable[0], replacement);
+
+    }
+
+    private static CharsetEncoder createEncoder(Charset charset) {
+        try {
+            return charset.newEncoder();
+        } catch (UnsupportedOperationException _) {
+            System.err.println("Could not create the character encoder!");
+        }
+        return null;
+    }
+
+    private static char[] findUnmappable(CharsetEncoder encoder) {
+
+        // Try to find a single-byte unmappable
+        Predicate<char[]> isUnmappable1 = createUnmappableSingleCharPredicate(encoder);
+        char[] unmappable1 = {0};
+        for (int i = 0; i < 0xFF; i++) {
+            unmappable1[0] = (char) i;
+            boolean unmappable = isUnmappable1.test(unmappable1);
+            if (unmappable) {
+                return unmappable1;
+            }
+        }
+
+        System.err.println("Could not find an unmappable character!");
+        return null;
+
+    }
+
+    static Predicate<char[]> createUnmappableSingleCharPredicate(CharsetEncoder encoder) {
+        return switch (encoder) {
+            case SingleByte.Encoder sbEncoder -> c -> sbEncoder.encode(c[0]) == UNMAPPABLE_ENCODING;
+            case DoubleByte.Encoder dbEncoder -> c -> dbEncoder.encodeChar(c[0]) == UNMAPPABLE_ENCODING;
+            default -> createUnmappableDoubleCharPredicate(encoder);
+        };
+    }
+
+    static Predicate<char[]> createUnmappableDoubleCharPredicate(CharsetEncoder encoder) {
+        assertEquals(encoder.unmappableCharacterAction(), CodingErrorAction.REPORT);
+        CharBuffer charBuffer = CharBuffer.allocate(2);
+        ByteBuffer byteBuffer = ByteBuffer.allocate(4);
+        return unmappable -> {
+            charBuffer.clear().put(unmappable).flip();
+            byteBuffer.clear();
+            CoderResult coderResult = encoder.encode(charBuffer, byteBuffer, true);
+            return coderResult.isUnmappable();
+        };
+    }
```

Comment:
I'd appreciate it if you can double-check these _"Is the given `char[]` unmappable for a particular encoder?"_ test generators.

Reply:
You're right. Simplified as suggested in ba1e5ff4de6.

I wanted to use `{Single,Double}Byte.Encoder` methods deliberately, since those were the ones triggering the failure. Though `CharsetEncoder#canEncode(char)` should do the trick, plus, it is public.